### PR TITLE
Explicit error handling on file ops

### PR DESCRIPTION
**Pull request by cbm755**
I found a bunch of `fopen`, `fputs`, `fclose` with no error handling.  So add that.

### CODE REVIEW

File: inst/private/python_ipc_system.m [hunk 1]
```diff
@@ -113,12 +113,21 @@
         disp (['This session will use the temp file "' tmpfilename '"'])
       end
     end
-    fd = fopen (tmpfilename, 'w');
+    [fd, msg] = fopen (tmpfilename, 'w');
+    if fd < 0
+      error (msg)
+    end
```
Comment (alexvong1995) [inline inst/private/python_ipc_system.m:119]: Should we prepend `system ipc:` to the error message and show the temp file name as well?

```
      error ('system ipc: failed to create temporary file %s: %s',
             tmpfilename, msg);
```

File: inst/private/python_ipc_system.m [hunk 2]
```diff
@@ -113,12 +113,21 @@
...
-    fputs(fd, bigs);
-    fclose(fd);
+    status = fputs(fd, bigs);
+    if status < 0
+      error ('failed to write the big string to temporary file')
+    end
+    status = fclose(fd);
+    if status < 0
+      error ('failed to close temporary file after writing')
+    end
```
Comment (alexvong1995) [inline inst/private/python_ipc_system.m:130]: Similarly, we can prepend `system ipc:` to the error message and show the temp file fd here (if needed).

**Review by alexvong1995 — COMMENTED** [review]
should be all good otherwise!

**Comment by cbm755** [comment]
Thanks @alexvong1995.

Will merge when pipeline succeeds

**Pull request merged**